libbeat/common/kubernetes/watcher.go

```diff
@@ -137,6 +139,8 @@ func NewWatcher(client kubernetes.Interface, resource Resource, opts WatchOption
 		UpdateFunc: func(o, n interface{}) {
 			if opts.IsUpdated(o, n) {
 				w.enqueue(n, update)
+			} else if opts.HonorReSyncs {
+				w.enqueue(n, add)
```

Comment:
Can you also please elaborate why in this case we push the event into the `add` queue and not in the `update`? Maybe add a comment about it for the future reader?

Reply:
i have done the needful